experiments/standalone/Soil/richards_runoff.jl

```diff
@@ -0,0 +1,246 @@
+    "processed_topographic_index 2.5 deg",
+    ArtifactFile[ArtifactFile(
+        url = "https://caltech.box.com/shared/static/dwa7g0uzhxd50a2z3thbx3a12n0r887s.nc",
+        filename = "means_2.5_new.nc",
+    ),],
+)
+infile_path = joinpath(get_data_folder(topmodel_dataset), "means_2.5_new.nc")
+regrid_dirpath = "regrid"
+outfile_root =
+    joinpath(pkgdir(ClimaLand), "experiments/standalone/Soil/static_data_cgll")
+ClimaLand.Regridder.hdwrite_regridfile_rll_to_cgll(
+    FT,
+    regrid_dirpath,
+    infile_path,
+    ["fmax", "landsea_mask"],
+    surface_space,
+    outfile_root;
+    mono = true,
+)
+
+file_info = ClimaLand.FileReader.FileInfo(
+    infile_path,
+    regrid_dirpath,
+    ["fmax", "landsea_mask"],
+    outfile_root,
+    [],
+    [],
+)
+data = ClimaLand.FileReader.PrescribedDataStatic{typeof(file_info)}(file_info)
+f_max = ClimaLand.FileReader.get_data_at_date(data, surface_space, "fmax")
+mask =
+    ClimaLand.FileReader.get_data_at_date(data, surface_space, "landsea_mask")
+oceans_to_zero(field, mask) = mask > 0.5 ? field : 0
+f_over = FT(3.28) # 1/m
+R_sb = FT(1.484e-4 / 1000) # m/s
+runoff_model = ClimaLand.Soil.Runoff.TOPMODELRunoff{FT}(;
+    f_over = f_over,
+    f_max = f_max,
+    R_sb = R_sb,
+)
+vg_α = ClimaCore.Fields.ones(subsurface_space) .* FT(0.2)
+hydrology_cm = map(vg_α) do (α)
+    FT = typeof(α)
+    ClimaLand.Soil.vanGenuchten{FT}(; α = α, n = α + FT(2))
+end
+θ_r = ClimaCore.Fields.zeros(subsurface_space)
+ν = ClimaCore.Fields.zeros(subsurface_space) .+ FT(0.5)
+K_sat = ClimaCore.Fields.zeros(subsurface_space) .+ FT(1e-6)
+S_s = ClimaCore.Fields.zeros(subsurface_space) .+ FT(1e-3)
+
+soil_params = ClimaLand.Soil.RichardsParameters(;
+    hydrology_cm = hydrology_cm,
+    ν = ν,
+    K_sat = K_sat,
+    S_s = S_s,
+    θ_r = θ_r,
+)
+
+function precip_function(t)
+    return -1e-6
+end
+precip = ClimaLand.TimeVaryingInput(precip_function)
+atmos = ClimaLand.PrescribedPrecipitation{FT, typeof(precip)}(precip)
+bottom_bc = ClimaLand.Soil.WaterFluxBC((p, t) -> 0.0)
+bc = (;
+    top = ClimaLand.Soil.RichardsAtmosDrivenFluxBC(atmos, runoff_model),
+    bottom = bottom_bc,
+)
+model = ClimaLand.Soil.RichardsModel{FT}(;
+    parameters = soil_params,
+    domain = domain,
+    boundary_conditions = bc,
+    sources = (),
+    lateral_flow = false,
+)
+Y, p, t = initialize(model)
+z = ClimaCore.Fields.coordinate_field(domain.space.subsurface).z
+lat = ClimaCore.Fields.coordinate_field(domain.space.subsurface).lat
+function hydrostatic_profile(
+    lat::FT,
+    z::FT,
+    ν::FT,
+    θ_r::FT,
+    α::FT,
+    n::FT,
+    S_s::FT,
+    fmax,
+) where {FT}
+    m = 1 - 1 / n
+    zmin = FT(-50.0)
+    zmax = FT(0.0)
+
+    z_∇ = FT(zmin / 5.0 + (zmax - zmin) / 2.5 * (fmax - 0.35) / 0.7)
+    if z > z_∇
+        S = FT((FT(1) + (α * (z - z_∇))^n)^(-m))
+        ϑ_l = S * (ν - θ_r) + θ_r
+    else
+        ϑ_l = -S_s * (z - z_∇) + ν
+    end
+    return FT(ϑ_l)
+end
+t0 = 0.0
+tf = 360.0 * 10
+dt = 360.0 * 2
```

Comment:
dt = 12 minutes
tf-t0 = 2 hours?

Reply:
Ill make it a little longer. we have some issues with speed :( as you have noticed too